### PR TITLE
Add promisekit extension for TezosNodeClient.

### PR DESCRIPTION
**Pull request by keefertaylor**
Adds Promise methods for TezosNodeClient.

Also cleans up an issue where there was an unused `completion` argument passed to an RPC.

### CODE REVIEW

File: Extensions/PromiseKit/TezosNodeClient+Promises.swift [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+    return forgeSignPreapplyAndInjectOperation(
+      operation: undelegateOperatoin,
+      source: source,
+      keys: keys
+    )
+  }
+
+  /// Register an address as a delegate.
+  /// - Parameters:
+  ///   - recipientAddress: The address which will receive the balance.
```
Comment (jlaws) [inline Extensions/PromiseKit/TezosNodeClient+Promises.swift]: These parameters don't match the method

File: Extensions/PromiseKit/TezosNodeClient+Promises.swift [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+  }
+
+  /// Delegate the balance of an originated account.
+  ///
+  /// Note that only KT1 accounts can delegate. TZ1 accounts are not able to delegate. This invariant
+  /// is not checked on an input to this methods. Thus, the source address must be a KT1 address and
+  /// the keys to sign the operation for the address are the keys used to manage the TZ1 address.
+  ///
+  /// - Parameters:
+  ///   - recipientAddress: The address which will receive the balance.
```
Comment (jlaws) [inline Extensions/PromiseKit/TezosNodeClient+Promises.swift]: These parameters don't match the method

File: Extensions/PromiseKit/TezosNodeClient+Promises.swift [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+
+  /// Retrieve the address manager key for the given address.
+  public func getAddressManagerKey(address: String) -> Promise<[String: Any]> {
+    let rpc = GetAddressManagerKeyRPC(address: address)
+    return send(rpc)
+  }
+
+  /// Transact Tezos between accounts.
+  /// - Parameters:
+  ///   - balance: The balance to send.
```
Comment (jlaws) [inline Extensions/PromiseKit/TezosNodeClient+Promises.swift]: balance --> amount ?

File: Extensions/PromiseKit/TezosNodeClient+Promises.swift (hunk 1 above)
Comment (keefertaylor) [reply Extensions/PromiseKit/TezosNodeClient+Promises.swift]: Done

File: Extensions/PromiseKit/TezosNodeClient+Promises.swift (hunk 3 above)
Comment (keefertaylor) [reply Extensions/PromiseKit/TezosNodeClient+Promises.swift]: Done

File: Extensions/PromiseKit/TezosNodeClient+Promises.swift (hunk 2 above)
Comment (keefertaylor) [reply Extensions/PromiseKit/TezosNodeClient+Promises.swift]: Done

**Comment by keefertaylor** [comment]
Also resolved documentation issues in TezosNodeClient.swift.

**Comment by keefertaylor** [comment]
Ignoring code coverage since the normal TezosNodeClient doesn't have any test coverage either. Will address in a future PR.

**Pull request merged**